Removed duplicate "SETMessage" in "Link Name" Option, a obstruction and removed "Navi Text" -ASCII line

GoldenMariaNova opened the pull request:
I removed a _**ID 1001 : SetMessage**_ in "_**ReduxTextBytes - Link.Text**_" duplicate and had to remove a hardcoded gravestone in one of "**Remove Owl**" options choices.

This object prevented the player to get access to the hole with the three **Deku** salesmen as a child, due to hardcoded relation with the **Lake Hylia** Owl, which stayed unmovable with the actor gone.

It's only for the Child **Link**'s version of the map. Adult's one is fine.

Admentus64 commented:
Removing the duplicate textbox looks good.

The actor edits have some issues, so please resolve:
1. No need to call PrepareMap for a second time if it's the same scene, map and header between actor functions. You're still applying changes to the same location.
2. No need to call SaveLoadedMap if you are not switching the scene or map.
3. Please replace the params of the gravestone actor instead so that they can be pushed. Just like in the Graveyard map. If possible of course.

GoldenMariaNova commented:
> snips

1. **Done**.
2. **Done**.
3. Impossible. Replacing the actor also do nothing, since it _IS_ like the other similar actors in the graveyard, since it is tagged as "**Default**". This one seems frustratingly peculiar, as it is hardcoded to another actor "**43**" being loaded and unloaded. Unless it is also removed, it will always obstruct the passage.

Also, like the optional altering functions of certain options in the patcher, it should not be a big deal, as it's almost an unexplored map as a child, and it is the most buggy too in vanilla _(Clipping in the **Owl** actor when going out from the hole, getting warped to Hyrule even its **NOT** intentional if the **Owl** text is skipped too fast, slow framerate in that sector in certain emulators)_...

It really don't change much from the vanilla game, as the hole will be accessible like it always was.

Admentus64 commented:
I check it out tomorrow.

I go try and see which params can be used for the gravestone.

GoldenMariaNova commented:
> snips

Just as a reminder, I corrected a missing comma and removed -ASCII from Navi text, which was making the option basically non-functional.

That's it.

Admentus64 commented:
![image](https://github.com/Admentus64/Patcher64Plus-Tool/assets/8747034/03da6cc5-78f5-40d7-9236-17d63f9ee456)

Easy 😄

Admentus64 commented:
Issues 1 and 2 still aren't resolved.

Unnecessary semicolumn (;) for ending code line 1963.

About the ASCII removal... It should stay there, but if it's an issue it's likely an issue with the Text Editor which has to be resolved.

Admentus64 commented:
Oh? You closed it? That wasn't necessary. It was almost good to go.

Well, nevermind, I apply these changes myself.

GoldenMariaNova commented:
> snips

Line 1960 through 1974 are part of the same script, so the semi-colon was made for clarity.

Also, I tried what you told me.

GoldenMariaNova commented:
> 

It was a mistake.

I don't know why, but I have issues with **Github**. I hope I resolved it.

Admentus64 commented:
Alright then.

GoldenMariaNova commented:
> snips

At the risk of sounding noob-ish... How? I never replaced actors before.

Admentus64 commented:
```PrepareMap -Scene "Lake Hylia" -Map 0 -Header 0
RemoveObject -Name "Gravestone"
RemoveActor -Name "Gravestone"
SaveAndPatchLoadedScene```

This is the code you used to remove both the actor and object.
Line 1: You load the scene, map and header you want to adjust
Line 2: Object is removed
Line 3: Actor is removed
Line 4: Changes are saved and applied

You don't need to remove any objects and actors at all. Instead you want to use `ChangeBytes -Offset "CF8C84" -Values "1500"`. This is a byte patching function, not a scene editor function. It's used to change bytes in the ROM.

You see, the gravestone is hardcoded to not be pushable under these three conditions:
- In Lake Hylia
- Is Child
- Owl is not active

The ChangeBytes command I just shown alters the code for the gravestone object so that it returns false on the age check. Which means the check always fails, which means the gravestone is pushable without a owl on top of it. That's how the gravestone can be kept and still be pushable.

Admentus64 commented:
Approved and ready for merge.
